### ISSUE TITLE
Attach new scheduler process in already running process

### ISSUE THREAD

**Issue by Nostritius** [skypjack/entt#845]
I have a problem with the entt::scheduler class. I have a timer process, which executes a script after a certain time. This script then attaches a new process to the scheduler. When this all happens during a scheduler update, it seems, that the new process is not properly attached and won't get executed. Here is a simple example which hopefully illustrates my problem:
```c++
#include <entt/entt.hpp>
#include <iostream>

class p1 : public entt::process<p1, float> {
public:
	p1() = default;

	void update(float delta, void *) {
		std::cout << "p1 started" << std::endl;
	}
};

class p2 : public entt::process<p2, float> {
public:
	p2(entt::scheduler<float> &scheduler) : _scheduler(scheduler) {}

	void update(float delta, void *) {
		_scheduler.attach<p1>();
	}

private:
	entt::scheduler<float> &_scheduler;
};

int main() {
	entt::scheduler<float> s;

	s.attach<p2>(s);

	while (true) {
		s.update(0);
		std::cout << "Current Processes " << s.size() << std::endl;
	}

	return 0;
}
```
Process p2 won't get executed and the number of current processes stays at 1. Is this a bug or expected behavior?

**Comment by skypjack** [comment]
Looking into it soon. In the meantime, isn't [this](https://godbolt.org/z/8q9EKfjb3) what you're looking for?

**Comment by Nostritius** [comment]
My current timer implementation is [here](https://github.com/OpenAWE-Project/OpenAWE/blob/master/src/timerprocess.cpp).The problem is, that a timer in this context is more generic and you do not know what will happen after it expired, since it executes bytecode. For this reason I cannot use then() at creation time. In my case I created a script function which starts an animation (Just a rotating sign), which is constantly updated over another process type. This process is started in the appropriate script function which will eventually executed over _bytecode->run(...).

**Comment by skypjack** [comment]
Interesting. So, from what I see in the code it's _expected behavior_ but I'm not particularly fond of it.
Briefly, it moves all the processes that die to the end of a vector, then erases them at once. If you add a new process, it goes to the end with them and therefore... 🤷‍♂️ 
Imho, we can change this behavior and drop processes in place when needed. It would make more sense and also introduce the support for your case automatically.

**Comment by skypjack** [comment]
Yeah, I did a super quick test. the change is actually easy and has literally no impact.
I'll add some tests and push it tomorrow on the `wip` branch. It would be great if you could confirm that it works as expected then.

**Comment by Nostritius** [comment]
Of Course, I will watch the wip branch and test it when it is ready
